src/ddspdrum/module.py

```diff
@@ -365,12 +366,27 @@ def __init__(
         self.amp_envelope = amp_adsr(sustain_duration)
         self.vco = vco
         self.vca = vca
+        self.noise_ratio = noise_ratio
 
     def __call__(self):
         self.pitch_envelope = fix_length(self.pitch_envelope, len(self.amp_envelope))
-        vco_out = self.vco(self.pitch_envelope)
+
+        vco_out = self.play_vco_bank(self.vco, self.pitch_envelope)
+        vco_out = self.add_noise(vco_out, self.noise_ratio)
+
         return self.vca(self.amp_envelope, vco_out)
 
+    @staticmethod
+    def play_vco_bank(vco_bank, pitch_envelope):
+        audio_out = 0
+        for vco in vco_bank:
+            audio_out += vco(pitch_envelope)
+        return audio_out / len(vco_bank)
+
+    @staticmethod
+    def add_noise(audio_in, ratio):
```

Comment:
Do we want to have noise as its own module?

Reply:
I'm fine to keep it here for now and refactor once we make a second synth besides Drum. But we can also refactor now.

*I imagined that we would have a collection of building block dsp modules, and then some preset architectures based on existing drum machines.* Agreed, different architectures but each one fixed.

Reply:
I'm making a Mixer module that combines two VCOs and optionally adds noise. I've seen modular units like this and it seems to make sense for now.

Reply:
@maxsolomonhenry So I'm against this for now unless we implement it correctly.

Right now, we assume all synth modules are atomic, i.e. don't contain other synth modules.

And the synths like Drum are a list of atomic synth modules (none of them compound)

If we start adding compound synth modules, it becomes more tricky to:
* read and set parameters for a whole synth
* iterate over all modules to make sure they are wired up well (have same sample and control rates)

I argue that unless we have a careful plan for how to do these things, we should not have compound modules for now. At least until we have Synth param setting and getting over a list of atomic modules